examples/dao/test/failing-path.js

```diff
@@ -0,0 +1,585 @@
+const APP_NOT_FOUND = 'RUNTIME_ERR1306: Application Index 1 not found or is invalid';
+
+describe('DAO - Failing Paths', function () {
+  let master, creator, proposer, voterA, voterB;
+  let depositLsigAcc, daoFundLsigAcc, proposalLsigAcc;
+  let ctx;
+
+  function setUpCtx () {
+    master = new AccountStore(1000e6);
+    creator = new AccountStore(initialBalance);
+    proposer = new AccountStore(initialBalance);
+    voterA = new AccountStore(initialBalance);
+    voterB = new AccountStore(initialBalance);
+    depositLsigAcc = new AccountStore(initialBalance); // runtime.account of depositLsig.address()
+    daoFundLsigAcc = new AccountStore(initialBalance);
+    proposalLsigAcc = new AccountStore(initialBalance);
+
+    ctx = new Context(
+      master, creator, proposer, voterA, voterB,
+      depositLsigAcc, daoFundLsigAcc, proposalLsigAcc
+    );
+  }
+
+  describe('Add proposal', function () {
+    this.beforeAll(setUpCtx);
+
+    let proposalParams, addProposalTx;
+    this.beforeEach(() => {
+      proposalParams = [
+        'str:add_proposal',
+        'str:my-custom-proposal', // name
+        'str:www.myurl.com', // url
+        'str:url-hash', // url_hash
+        'str:', // hash_algo (passing null)
+        `int:${votingStart}`, // votingStart (now + 1min)
+        `int:${votingEnd}`, // votingEnd (now + 3min)
+        `int:${executeBefore}`, // executeBefore (now + 7min)
+        `int:${ProposalType.ALGO_TRANSFER}`, // type
+        `addr:${ctx.daoFundLsig.address()}`, // from (DAO treasury)
+        `addr:${ctx.proposer.address}`, // recepient
+        `int:${2e6}` // amount (in microalgos)
+      ];
+
+      addProposalTx = mkProposalTx(
+        ctx.daoAppID,
+        ctx.govTokenID,
+        ctx.proposer.account,
+        ctx.depositLsig,
+        ctx.proposalLsig,
+        ctx.daoFundLsig
+      );
+      addProposalTx[1].amount = 15;
+    });
+
+    it('should fail if proposalLsig not opted-in to DAO App', () => {
+      assert.throws(() => ctx.executeTx(addProposalTx), APP_NOT_FOUND);
+    });
+
+    it('should fail if votingEnd < votingStart', () => {
+      ctx.optInToDAOApp(ctx.proposalLsig.address()); // opt-in
+      ctx.syncAccounts();
+
+      // set votingEnd < votingStart (30s before)
+      proposalParams[6] = `int:${votingStart - 30}`;
+      addProposalTx[0].appArgs = proposalParams;
+
+      assert.throws(() => ctx.executeTx(addProposalTx), 'RUNTIME_ERR1005: Result of current operation caused integer underflow');
+    });
+
+    it('should fail if "min_duration <= votingEnd - votingStart <= max_duration" is not satisfied', () => {
+      // increase votingEnd (voting duration set as 9min, but allowed is 5min)
+      proposalParams[6] = `int:${votingEnd + (10 * 60)}`;
+      addProposalTx[0].appArgs = proposalParams;
+
+      assert.throws(() => ctx.executeTx(addProposalTx), RUNTIME_ERR1009);
+    });
+
+    it('should fail if executeBefore < votingEnd', () => {
```

Comment:
it should also fail when `executeBefore == votingEnd`

Reply:
done